paper-input-behavior.html

```diff
@@ -302,9 +302,27 @@
     },
 
     attached: function() {
+      this._checkForNativeFormatter();
       this._updateAriaLabelledBy();
     },
-
+    
+    // set alwaysFloatLabel to true if browser has native formatter (input->type:date or input->type:time etc.)
+    _checkForNativeFormatter: function() {
+      // only check these types if the label has to be always floated.
+      var checkTypes = ['number', 'time', 'date'];
+      if (checkTypes.indexOf(this.inputElement.type) === -1) {
+        return;
+      }
+      // TODO make value change silent
+      var value = this.inputElement.value;
+      // TODO catch chrome format warning
+      this.inputElement.value = ' ';
```

Comment:
Basically now every time you're attaching this input you're changing the value at least twice (once to empty, and once to the original value). Each of these has a bunch of side effects, and fires a bunch of observers, so it's a really bad approach when it comes to performance